src/taskpane/server.js

```diff
@@ -92,25 +104,32 @@ var API_PROJECT_BASE = '/services/v6_0/RestService.svc/projects/',
     someError: 2,
     allError: 3,
     wrongSheet: 4,
-    existingEntries: 5
+    existingEntries: 5,
+    advancedMode: 6,
+    noData: 7
   },
   SUBTYPE_IDS = ["TestCaseId", "TestStepId"],
   STATUS_MESSAGE_GOOGLE = {
     1: "All done! To send more data over, clear the sheet first.",
     2: "Sorry, but there were some problems (see the cells marked in red). Check any notes on the relevant ID field for explanations.",
-    3: "We're really sorry, but there was a problem while sending data to Spira. Some records may had be exported to Spira. Please check it on the application.",
+    3: "We're really sorry, there was a problem sending data to Spira. Some records may not have been sent correctly. Please check in Spira to confirm.",
     4: "You are not on the correct worksheet. Please go to the sheet that matches the one listed on the Spira taskpane / the selection you made in the sidebar.",
-    5: "Some/all of the rows already exist in SpiraPlan. These rows have not been re-added."
+    5: "Some/all of the rows already exist in SpiraPlan. These rows have not been re-added.",
+    6: "Data sent! Since you are using the advanced fields mode, any errors related to comments and/or associations will be marked as red in the spreadsheet. To send more data over, clear the sheet first.",
+    7: "No records were returned from Spira. Please verify the page number, product, and artifact. If the problem persists, contact your Spira administrator.",
+    8: "There was an error processing your request. Please make sure that all the required fields are populated and the current user has access to the selected Template. Also, make sure the selected Template has at least one (1) active product."

```

Comment:
This seems fairly similar to error 3 but more descriptive, is there a reason these both need to exist?

Reply:
Actually, yes. This was the solution for [IN:7599], in a different scenario (no products for template). We agreed it was better to make it very clear for the user